shell/platform/darwin/common/framework/Source/FlutterChannels.mm

```diff
@@ -9,12 +9,28 @@
 #pragma mark - Basic message channel
 
 static NSString* const kFlutterChannelBuffersChannel = @"dev.flutter/channel-buffers";
+static NSString* const kResizeMethod = @"resize";
+static NSString* const kOverflowMethod = @"overflow";
 
 static void ResizeChannelBuffer(NSObject<FlutterBinaryMessenger>* binaryMessenger,
                                 NSString* channel,
                                 NSInteger newSize) {
-  NSString* messageString = [NSString stringWithFormat:@"resize\r%@\r%@", channel, @(newSize)];
-  NSData* message = [messageString dataUsingEncoding:NSUTF8StringEncoding];
+  NSArray* args = @[ channel, [NSNumber numberWithInt:newSize] ];
```

Comment:
IIRC the Google style guide used to mention it; most likely it was culled under https://google.github.io/styleguide/objcguide.html#style-rules-should-pull-their-weight because there's no reason *not* to use the modern literal syntax.

Boxing with NSNumber should definitely use modern boxing syntax.

Reply:
When switching to the modern boxing syntax, for this particular code, it does not produce the same result at runtime.
As you can see in the change I pushed, the test is failing because the serialization result is not the same (and the result is wrong with the modern boxing syntax). I don't have enough ObjectiveC knowledge to understand why.

Test failure: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8770883490087443713/+/u/test:_Tests_for_ios_debug_sim_arm64_extension_safe__3_/stdout

Failure message: 
```console
error: -[FlutterChannelsTest testResize] : OCProtocolMockObject(FlutterBinaryMessenger): unexpected method invoked: sendOnChannel:@"dev.flutter/channel-buffers" message:{length = 33, bytes = 0x07067265 73697a65 0c02070c 666c7574 ... 03000000 00000000 } 
expected:	sendOnChannel:@"dev.flutter/channel-buffers" message:{length = 29, bytes = 0x07067265 73697a65 0c02070c 666c7574 ... 65737403 03000000 } (NSInternalInconsistencyException)
```

Reply:
The failing test says that the expected value was created with Dart code; why? It's not clear to me why you are asserting a specific representation here that comes from a different implementation with different integer semantics.

Reply:
The encoded message is deserialized in the Dart side, the deserialization code is there:

https://github.com/flutter/engine/blob/93e8901490e78c7ba7e319cce4470d9c6478c6dc/lib/ui/channel_buffers.dart#L459-L537

The test checks that the message that will be send to Dart as an array of bytes is exactly the expected one.
With the old boxing syntax (`NSArray* args = @[ channel, [NSNumber numberWithInt:newSize] ];`), calling `encodeMethodCall` returns the exact byte array. 

The problem is not specific to the test, when running a Flutter app with a modified engine that calls the resize API, the Dart side throws the following exception:
```console
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Exception: Invalid arguments for 'resize' method sent to dev.flutter/channel-buffers (second argument must be an integer in the range 0 to 2147483647)
#0      ChannelBuffers.handleMessage (dart:ui/channel_buffers.dart:496:13)
#1      PlatformDispatcher._dispatchPlatformMessage (dart:ui/platform_dispatcher.dart:674:24)
#2      _dispatchPlatformMessage (dart:ui/hooks.dart:257:31)
```

Based on the Dart exception, I figured out that the modern boxing syntax is treating the NSInteger as a long on 64bits platforms. I updated the PR with a cast to int and the encoded message is ok now.

Reply:
I have updated the PR with a comment on the cast.